scripts: add `gceworker.sh update-firewall`

tbg opened the pull request:
This makes it easier to allowlist the current IP and also adds a prompt
in `gceworker.sh start` that explains why such allowlisting is
necessary, thus making it discoverable to those that would previously
wonder about the opaque timeouts that would occur in the absence of an
appropriate firewall rule.

Firewall rules are not updated automatically, as such an
action is only appropriate when at home or at an office.

For more context, see:

https://cockroachlabs.slack.com/archives/C0HM2DZ34/p1719878745576399

Epic: none
Release note: None

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/130017)
<!-- Reviewable:end -->

rickystewart commented on scripts/gceworker.sh:
```diff
@@ -19,6 +19,11 @@ fi
+    if [ -z "${GCEWORKER_NO_FIREWALL_UPDATE-}" ]; then
+	# https://cockroachlabs.atlassian.net/wiki/spaces/devinf/pages/65077811/Building+CRDB+with+gceworker#Add-Your-IP-to-the-GCE-SSH-Allow-List
+        echo "updating firewall rule (disable with GCEWORKER_NO_FIREWALL_UPDATE=true)..."
+        $0 update-firewall
```
Nit: preference to put this in a `function` rather than re-call the script with the `update-firewall` argument

tbg committed:
scripts: add `gceworker.sh update-firewall`

This makes it easier to allowlist the current IP and also adds a prompt
in `gceworker.sh start` that explains why such allowlisting is
necessary, thus making it discoverable to those that would previously
wonder about the opaque timeouts that would occur in the absence of an
appropriate firewall rule.

Firewall rules are not updated automatically, as such an
action is only appropriate when at home or at an office.

For more context, see:

https://cockroachlabs.slack.com/archives/C0HM2DZ34/p1719878745576399

Epic: none
Release note: None

rickystewart commented on scripts/gceworker.sh:
```diff
@@ -19,6 +19,17 @@ fi
     gcloud compute instances start "${1}"
+    if [ -z "${GCEWORKER_NO_FIREWALL_WARNING-}" ]; then
+	      cat <<EOF
+	      As of July 2024, due to an SSH vulnerability, access to gceworkers is permitted only from home and office IPs[1].
```
What if you printed the warning only if SSH fails? At this point, most people who regularly use GCEWorkers must have the firewall rules set up. If you do it so that the warning only happens if the SSH fails, then you avoid spamming those people who are already on top of it.

tbg replied:
Not a bad thought - but many work from home and from the office, but we only get one firewall rule. Don't want to spend my cycles on diminishing returns.

tbg commented:
Reworded the message slightly. TFTR! Merging as per DM.

bors r+
